src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ReadyToRunHelperNode.cs

```diff
@@ -133,5 +156,75 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
                 return null;
             }
         }
+
+        protected override void OnMarked(NodeFactory factory)
+        {
+            switch (_id)
+            {
+                case ReadyToRunHelperId.GenericLookupFromThis:
+                case ReadyToRunHelperId.GenericLookupFromDictionary:
+                    {
+                        // When the helper call gets marked, ensure the generic layout for the associated dictionaries
+                        // includes the signature.
+                        var lookupInfo = (GenericLookupDescriptor)_target;
+                        factory.GenericDictionaryLayout(lookupInfo.CanonicalOwner).EnsureEntry(lookupInfo.Signature);
```

Comment:
I guess it can allow us to do sparsely populated dictionaries ... but it is an optimization that can be done later.